### PR TITLE
#230 - reproduce FileAlreadyExistsException

### PR DESCRIPTION
**Pull request by olenagerasimova**
Added test to reproduce #230 and todo to fix.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#234](https://github.com/artipie/rpm-adapter/issues/234) is assigned to @g4s8[/z](https://www.0crat.com/u/g4s8), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/0d5b78c5-35a7-4d15-acb5-0446a1d80370); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by g4s8** [comment]
@olenagerasimova to catch such issues, I ran tests for ARtipie adapters using JDK1.8 - they have different behavior comparing to JDK11 for generic representation (when using generic in generic):
 - JDK1.8
   - `List<String> lst = new ArrayList<>()` - OK
   - `List<Map.Entry<String, String>> lst = new ArrayList<>()` - not OK
 - JDK11
   - `List<String> lst = new ArrayList<>()` - OK
   - `List<Map.Entry<String, String>> lst = new ArrayList<>()` - OK

**Comment by olenagerasimova** [comment]
@g4s8 could you please help me here? Do have any idea how to make the test stable on windows?

File: src/test/java/com/artipie/rpm/RpmTest.java [hunk 1]
```diff
@@ -48,6 +62,64 @@
...
+    @Test
+    void updatesSameRepoSimultaneously() throws Exception {
+        final Storage storage = new InMemoryStorage();
+        final Rpm repo =  new Rpm(
+            storage, StandardNamingPolicy.SHA1, Digest.SHA256, true
+        );
+        RpmTest.putFilesInStorage(storage);
+        final Scalar<Boolean> task = new Unchecked<>(
+            () -> {
+                repo.batchUpdate(Key.ROOT).blockingAwait();
```
Comment (g4s8) [inline src/test/java/com/artipie/rpm/RpmTest.java:101]: @olenagerasimova I'd use `CountDownLatch` here to be sure that all threads are waiting before starting update

**Comment by olenagerasimova** [comment]
@g4s8 compilation error is not the problem, test is unstable as we have to call `MetadataFile#save` at almost the same time from different threads. It does reproduce on linux and mac stable though...

**Comment by olenagerasimova** [comment]
@g4s8 take a look please

**Comment by olenagerasimova** [comment]
@g4s8 will you merge it, or maybe I better close this PR and submit new one with the fix?

**Comment by g4s8** [comment]
> @g4s8 will you merge it, or maybe I better close this PR and submit new one with the fix?

@olenagerasimova yes, I'd fix it without test, since the tests are not working yet

**Comment by 0crat** [comment]
Job was finished in 12 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Pull request [#234](https://github.com/artipie/rpm-adapter/issues/234) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Pull request closed without merging**